cmd/observe/flows_filter.go

```diff
@@ -337,37 +469,38 @@ func (of *flowFilter) set(f *filterTracker, name, val string, track bool) error
 		})
 	// label filters
 	case "label":
+		if strings.HasPrefix(val, "k8s:io.kubernetes.pod.namespace=") {
```

Comment:
How about also handling the `io.kubernetes.pod.namespace=` prefix (i.e. without the `k8s:` part)? I guess most users don't provide the (cilium specific) [`k8s:` label source](https://github.com/cilium/cilium/blob/33573bd0e7e797e25d81ab183e742cb348729d95/pkg/labels/labels.go#L108-L109), and IMHO it's safe enough to assume that this label will and should only be defined by k8s. Thoughts?

Reply:
Maybe I'm holding it wrong, but the `io.kubernetes.pod.namespace` label isn't set in my dev environment, all k8s labels have the `k8s:` prefix. So if I for example run `hubble observe --label io.kubernetes.pod.namespace=kube-system` I don't get any flows at all.

Should this actually work?

Reply:
So looking into this I realized two things:

1. Yes, this should work but doesn't. There is a server side bug in the labels filter, where hubble doesn't properly handle labels with dots `.` and no prefix with a colon `:`. So that means `hubble observe --label k8s-app=hubble-relay`  works as expected but `hubble observe --label io.kubernetes.pod.namespace=kube-system` doesn't. The second one only matches a `io:kubernetes.pod.namespace` label. For now I'd assume that this works for this PR and will look into fixing this in `cilium/cilium`. 
2. My simplistic prefix label parsing is broken. If we try to use more complex label selector like `hubble observe --label k8s:io.kubernetes.pod.namespace=cilium,k8s:io.cilium in (is-awesome,rocks)`, we suddenly try to filter for pods in the namespace `cilium,k8s:io.cilium in (is-awesome,rocks)`. Doing this "properly" for more complex label selectors is actually not trivial

I'm starting to question if the whole special treatment for the namespace label actually makes sense. Doing this "properly" for any label selector IMO introduces a lot of complexity for not much gain and doing it for "simple" label selectors only (i.e. only if the selector is of the form `io.kubernetes.pod.namespace=<namespace>`) seems a bit magical and I think as a user it's not entirely clear for which selectors this will or won't work.

Lastly, who is actually going to write `hubble observe --label io.kubernetes.pod.namespace=kube-system get pod foo` instead of `hubble observe -n kube-system get pod foo` and is it worth it to special case this?

Reply:
@glrf all fair questions, let's drop the specific treatment of the namespace label!